velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -945,7 +947,9 @@ struct DateTruncFunction : public TimestampWithTimezoneSupport<T> {
     auto dateTime = getDateTime(date);
     adjustDateTime(dateTime, unit);
 
-    result = timegm(&dateTime) / kSecondsInDay;
+    int64_t seconds;
+    VELOX_CHECK(Timestamp::utcToEpoch(dateTime, seconds));
+    result = seconds / kSecondsInDay;
```

Comment:
for example, this would be:

```
result = Timestamp::utcToEpoch(dateTime, seconds) / kSecondsInDay
```

assuming you don't need the assert since your static_assert will give you a compile-time error.